### DISCUSSION TITLE
Morbo for Minion

### DISCUSSION THREAD

**Discussion by s1037989** [mojolicious/mojo#2198] [Q&A]
Is there a way to run a Minion worker like morbo -- when the code base changes, the worker auto restarts?  Or is there a different way I should be running a Minion worker during development?

**Comment by kraih** [comment]
I think so far there are only bad solutions. Closest i've ever come to using something like that in development was a recurring timer in the app process that calls `$minion->perform_jobs` or `$minion->perform_jobs_in_foreground` every few seconds.

**Comment by s1037989** [comment]
Thanks!  I ended up doing this so that if there are no registered workers (when doing development) then the main app will perform the jobs but also log a warning that it's doing so if the app mode is production.  Seems a decent way to go.

```
  Mojo::IOLoop->recurring(3 => sub {
    my $minion = $app->minion;
    my $workers = $minion->workers;
    return if $workers->total;
    my $inactive = $minion->jobs({states => ['inactive']})->total;
    return unless $inactive;
    $app->log->warn("No registered workers, performing $inactive jobs in app") if $app->mode eq 'production';
    $minion->perform_jobs;
  });
```

**Comment by rabbiveesh** [comment]
There's a tool on cpan called App::lrrr, it's literally the morbo restart
engine which restarts any arbitrary program passed in on the command line.
I have been happily using it for several years

**Comment by s1037989** [comment]
That's excellent!  Thanks for bringing awareness to that!
